### PR TITLE
Adding Rules API to ServiceBus swagger 

### PR DESCRIPTION
**Pull request by sazeesha**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 
### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).
### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [ ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

---

This change is [Reviewable](https://reviewable.io/reviews/azure/azure-rest-api-specs/629)

### CODE REVIEW

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 1]
```diff
@@ -2642,6 +2889,196 @@
...
+      "description": "Represents the filter actions which are allowed for the transformation of a message that have been matched by a filter expression."
+    },
+	"SqlRuleAction": {
+      "type": "object",
+      "properties": {
+         "SqlExpression": {
+          "type": "string",
+          "description": "Gets the SQL expression."
+        },
+		"CompatibilityLevel": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: incorrect indentation. make sure that indentation is 2 spaces

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 2]
```diff
@@ -1858,6 +1858,253 @@
...
+            "required": true,
+            "type": "string",
+            "description": "The topic name."
+          },
+          {
+            "name": "subscriptionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The subscriptionName name."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Better description. It gets confusing with subscriptionId and subscriptionName

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 3]
```diff
@@ -1858,6 +1858,253 @@
...
+            "description": "The rule name."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RuleCreateOrUpdateParameters"
+            },
+            "description": "Parameters supplied to create a rules."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: "Parameters supplied to create a rule~~s~~.

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 4]
```diff
@@ -1858,6 +1858,253 @@
...
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Rules"
+        ],
+        "operationId": "Rules_Delete",
+        "description": "Deletes an existing rule.",
+        "externalDocs": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: why have this defined if it is empty?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 5]
```diff
@@ -2642,6 +2889,196 @@
...
+          "$ref": "#/definitions/RuleAction",
+          "description": "Represents the filter actions which are allowed for the transformation of a message that have been matched by a filter expression."
+        },
+        "CreatedAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Gets creation time of the rule."
+        }
+      },
+      "description": "Description of Subscription Resource."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: description seems incorrect

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 6]
```diff
@@ -2642,6 +2889,196 @@
...
+        "Filter": {
+          "$ref": "#/definitions/Filter",
+          "description": "Describes a filter expression that is evaluated against a BrokeredMessage."
+        },
+        "RuleAction": {
+          "$ref": "#/definitions/RuleAction",
+          "description": "Represents the filter actions which are allowed for the transformation of a message that have been matched by a filter expression."
+        },
+        "CreatedAt": {
+          "type": "string",
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: readOnly: true ?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 7]
```diff
@@ -2642,6 +2889,196 @@
...
+          "format": "date-time",
+          "description": "Gets creation time of the rule."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Remove the phrase Gets

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 8]
```diff
@@ -2642,6 +2889,196 @@
...
+      "properties": {
+         "RequiresPreprocessing": {
+          "type": "boolean",
+          "description": "Gets a value indicating whether the SQL filter expression requires preprocessing."
+        }
+      },
+      "description": "Represents set of CompositeAction that is performed against a ServiceBus.Messaging.BrokeredMessage "
+    },	
+    "RuleProperties": {
+      "properties": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: which properties are required in this object ?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 9]
```diff
@@ -2642,6 +2889,196 @@
...
+          "type": "boolean",
+          "description": "Gets a value indicating whether the CorrelationFilter expression requires preprocessing."
+        }
+      },
+      "description": "Represents the correlation filter expression."
+    },
+    "RuleAction": {
+      "type": "object",
+      "properties": {
+         "RequiresPreprocessing": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: what are the required set of properties?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 10]
```diff
@@ -2642,6 +2889,196 @@
...
+        }
+      },
+      "description": "Represents the filter actions which are allowed for the transformation of a message that have been matched by a filter expression."
+    },
+	"SqlRuleAction": {
+      "type": "object",
+      "properties": {
+         "SqlExpression": {
+          "type": "string",
+          "description": "Gets the SQL expression."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Can you provide an example in the description?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 11]
```diff
@@ -2642,6 +2889,196 @@
...
+          "type": "boolean",
+          "description": "Gets a value indicating whether the SQL filter expression requires preprocessing."
+        }
+		},
+      "description": "Represents set of actions written in SQL language-based syntax that is performed against a ServiceBus.Messaging.BrokeredMessage "
+      },
+	"CompositeAction": {
+      "type": "object",
+      "properties": {
+         "RequiresPreprocessing": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: is this a required property or an optional property? If optional then what is the default value? If there is a default then it should be defined.

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 12]
```diff
@@ -2642,6 +2889,196 @@
...
+		"format": "int32",
+		"type": "integer",
+          "description": "This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20."
+		}
+      },
+      "description": "Represents a filter which is a composition of an expression and an action that is executed in the pub/sub pipeline."
+    },	
+	"CorrelationFilter": {
+      "type": "object",
+      "properties": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: what are the required and readOnly properties of this object ?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 13]
```diff
@@ -2642,6 +2889,196 @@
...
+          "type": "string",
+          "description": "Gets the identifier of the correlation."
+        },
+		"MessageId": {
+          "type": "string",
+          "description": "Gets the identifier of the message."
+        },		
+		"To": {
+          "type": "string",
+          "description": "Gets the address to send to."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: should this be an email address or twitter handle? Can you provide an example in the description?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 14]
```diff
@@ -2642,6 +2889,196 @@
...
+		},
+		"SessionId": {
+		  "type": "string",
+          "description": "Gets the session identifier."
+		},
+		"ReplyToSessionId": {
+		  "type": "string",
+          "description": "Gets the session identifier to reply to."
+		},
+		"ContentType": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: enum candidate?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 15]
```diff
@@ -2642,6 +2889,196 @@
...
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Description of Rule Resource."
+    },
+    "Filter": {
+      "type": "object",
+      "properties": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: are there any required or any readOnly properties in this object?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 16]
```diff
@@ -2642,6 +2889,196 @@
...
+         "RequiresPreprocessing": {
+          "type": "boolean",
+          "description": "Preprocesses the filter expression and returns a preprocessed FilterExpression."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: description seems to be out of context

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 17]
```diff
@@ -2642,6 +2889,196 @@
...
+        }
+      },
+      "description": "MessageDescribes a filter expression that is evaluated against a BrokeredMessage."
+    },
+	"SqlFilter": {
+      "type": "object",
+      "properties": {
+         "SqlExpression": {
+          "type": "string",
+          "description": "The SQL expression."
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: example?

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 18]
```diff
@@ -2642,6 +2889,196 @@
...
+		"CorrelationFilter": {
+          "$ref": "#/definitions/CorrelationFilter",
+          "description": "Represents the correlation filter expression."
+        }
+      },
+      "description": "MessageDescribes a filter expression that is evaluated against a BrokeredMessage."
+    },
+	"SqlFilter": {
+      "type": "object",
+      "properties": {
```
Comment (amarzavery) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: are there any required or any readOnly properties in this object?

**Review by amarzavery — CHANGES_REQUESTED** [review]

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 2 above)
Comment (v-Ajnava) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Resolved.

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 3 above)
Comment (v-Ajnava) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Resolved

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 14 above)
Comment (sazeesha) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Not really, it is used to just match the whatever is in the brokered message.

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 13 above)
Comment (sazeesha) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: I will update it with a better description. It is used to match the To property of a brokered message

File: arm-servicebus/2015-08-01/swagger/servicebus.json [hunk 19]
```diff
@@ -2642,6 +2889,196 @@
...
+          "description": "Gets the SQL expression."
+        },
+		"CompatibilityLevel": {
+		"format": "int32",
+		"type": "integer",
+          "description": "This property is reserved for future use. An integer value showing the compatibility level, currently hard-coded to 20."
+		},
+		"RequiresPreprocessing": {
+          "type": "boolean",
+          "description": "Gets a value indicating whether the SQL filter expression requires preprocessing."
```
Comment (v-Ajnava) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Resolved

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 10 above)
Comment (v-Ajnava) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Resolved, example added.

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 1 above)
Comment (v-Ajnava) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Corrected indentation.

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 5 above)
Comment (v-Ajnava) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Resolved.

File: arm-servicebus/2015-08-01/swagger/servicebus.json (hunk 7 above)
Comment (v-Ajnava) [inline arm-servicebus/2015-08-01/swagger/servicebus.json]: Resolved.

**Pull request closed without merging**